eclipse-theia/theia
# Translate monaco using default keys

#### What it does

Now that https://github.com/eclipse-theia/theia/pull/10736 has been merged, our previous method of translating monaco doesn't work anymore. Instead it can be accomplished way easier :)

Before loading any monaco related code, we simply override the `localize` function to delegate to our `nls.localize` function. This doesn't work in all cases (210 of 4200 strings remain untranslated this way), but most of these aren't visible to users anyway. I've made the `getDefaultKey` function a bit more lenient to support labels which have only changed in their capitalization. For example VSCode changed from `Previous match` to `Previous Match`, which should still be translated correctly.

Also updates some generated `nls.localizeByDefault` calls which weren't working correctly. I updated the preference extractor for that. It seems like the eslint rule doesn't pick these up for some reason. Really weird :/

I will probably come back to this next month and translate the missing pieces as well.

#### How to test

0. Running Theia in English shouldn't affect any strings displayed by Monaco
1. Download and install a language pack of your choice
2. Open the editor and associated menus/widgets
3. Confirm that most of the user visible labels are translated.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
